src/components/TimeSeriesComponent/ElevationChart.vue

```diff
@@ -0,0 +1,328 @@
+
+interface Tag {
+  id: string;
+  name: string;
+  disabled: boolean;
+  legendSvg: string;
+}
+
+@Component
+export default class ElevationChart extends Vue {
+  @Prop({
+    default: () => {
+      return {}
+    }
+  })
+  value!: ChartConfig
+
+  @Prop({
+    default: () => {
+      return {}
+    }
+  })
+  series!: Record<string, Series>
+  axis!: CartesianAxes
+  isFullscreen = false
+  legendTags: Tag[] = []
+
+  created(): void {
+    this.legendTags = []
+  }
+
+  mounted(): void {
+    const axisOptions: CartesianAxesOptions = {
+      x: [{
+        type: AxisType.value,
+        position: AxisPosition.Bottom,
+        label: ' ',
+        unit: ' ',
+        nice: true,
+        showGrid: true
+      }],
+      y: [{
+        position: AxisPosition.Left,
+        showGrid: true,
+        label: ' ',
+        unit: ' ',
+        nice: true
+      },
+        {
+          position: AxisPosition.Right,
+          label: ' ',
+          unit: ' ',
+          nice: true
+        }
+      ],
+      margin: {
+        left: 80,
+        right: 80
+      }
+    }
+
+    this.setChartConfigValues(axisOptions)
+
+    const containerReference = this.$refs['elevation-chart-container'] as HTMLElement
+    this.axis = new CartesianAxes(containerReference, 800, 1200, axisOptions)
+    const mouseOver = new MouseOver()
+    const zoom = new ZoomHandler(WheelMode.X)
+    this.axis.accept(zoom)
+    this.axis.accept(mouseOver)
+    this.resize()
+    if (this.value !== undefined) this.onValueChange()
+    window.addEventListener('resize', this.resize)
+  }
+
+  @Watch('series', { deep: true})
+  @Watch('value')
+  onValueChange(): void {
+    this.clearChart()
+    this.refreshChart()
+    this.setTags()
+    this.$forceUpdate()
+  }
+
+  beforeDestroy(): void {
+    window.removeEventListener('resize', this.resize)
+  }
+
+  get fullscreenIcon(): string {
+    return this.isFullscreen ? 'mdi-fullscreen-exit' : 'mdi-fullscreen'
+  }
+
+  setChartConfigValues(axisOptions: CartesianAxesOptions): void {
+    const chartConfig = this.value
+    const yAxisOptions = chartConfig.yAxis
+    const xAxisOptions = chartConfig.xAxis
+
+    yAxisOptions?.forEach((axis, i) => {
+      axisOptions.y[i].type = axis.type as AxisType
+    })
+
+    xAxisOptions?.forEach((axis, i) => {
+      axisOptions.x[i].type = axis.type as AxisType
+    })
+  }
+
+  refreshChart(): void {
+    const ids: string[] = this.axis.charts.map((c: any) => c.id)
+    const removeIds: string[] = this.axis.charts.map((c: any) => c.id)
+    if (this.value?.series === undefined) return
+    for (const series of this.value.series) {
+      if (!series.visibleInPlot) continue
+      if (!ids.includes(series.id)) {
+        this.addToChart(series)
+      }
+      const index = removeIds.findIndex((item) => {
+        return item === series.id
+      })
+      if (index >= 0) removeIds.splice(index, 1)
+    }
+    for (const id of removeIds) {
```

Comment:
use the more compact way: `removeIds.forEach(id => this.axis.removeChart(id))`

Reply:
done